### DISCUSSION TITLE
Error with window.ethereum undefined

### DISCUSSION THREAD

**Discussion by RadimKotajnyDev** [vercel/next.js#44899] [Help] [answered]
Hello, i'm getting an error: "TypeError: undefined is not an object (evaluating 'window.ethereum.selectedAddress = undefined')", but i don't have idea why. Can it be some malicious code from NPM package or something?

**Comment by icyJoseph** [comment, marked as answer]
How did you get this error? Could you post some screenshots? Is it happening on incognito mode as well?

**Reply by RadimKotajnyDev** [reply]
Yup, same on incognito mode with turned off Brave Shields and adblock. BTW. Error is told by NextJS
![IMG-5541](https://user-images.githubusercontent.com/67335383/212561725-228fc8f6-7516-4cf8-a1fe-3d53c54a6c71.PNG)

**Reply by RadimKotajnyDev** [reply]
And this is my package.json: 
![obrazek](https://user-images.githubusercontent.com/67335383/212561972-95b8c882-3489-491f-b6a2-4184963062af.png)

**Reply by icyJoseph** [reply]
The thing is that Next.js catches errors that happen while rendering. So it is expected to trap this kind of problem, accessing window during SSR. The question is, who's trying to access window during SSR... I'll do a quick scan of those packages.

**Reply by icyJoseph** [reply]
https://github.com/localtunnel/localtunnel/issues/514 looks like you are not alone...

**Reply by icyJoseph** [reply]
I guess, this is a brave browser issue then? Can you try other browsers?

**Reply by RadimKotajnyDev** [reply]
It's just on brave. So it's safe to use? No malicious NPM packages?

**Reply by icyJoseph** [reply]
Yeah, could you describe your setup? I wanna see if I can trigger it myself.

You are not in danger.

**Reply by icyJoseph** [reply]
Just for completeness sake, as it was very late when I answered to this. 

Some browser environments, including extensions, run JS side by side with your app. Think of TikTok in-app browser, or Facebook's, actual browsers, like Chrome, Brave, etc, etc...

Next.js, during dev mode, has something called, `react-dev-overlay`, which attaches an `unhandledrejection` and `error` listeners, onto the window object. Should something fail and not be properly handled, it'll arrive to the `react-dev-overlay`.

- https://github.com/vercel/next.js/blob/canary/packages/react-dev-overlay/src/client.ts#L59

So when the `window.ethereum` line-of-code fails, since nothing handled that exception, it arrives to the developer-error-overlay. It doesn't mean there's a problem with Next.js. 

When I visit any site on Brave, I see that `window.ethereum` is a thing. For some reason in your setup, it is missing. In the Brave browser repositories, I could only find this Swift file which does `window.ethereum.selectedAddress`:

- https://github.com/brave/brave-ios/blob/41ddb78f1dd31a675d9d01ea969f115f8a911065/Client/Frontend/Browser/BrowserViewController/BrowserViewController%2BWallet.swift#L448

**Reply by RadimKotajnyDev** [reply]
Thanks a lot for explanation and help. I'm not sure what information you need for setup, but i'm using Next 12 (I will upgrade to 13), Win11, the packages in package.json,  and error is only in brave.

**Reply by icyJoseph** [reply]
Do you see the error going to other sites as well? Within Brave? At this point this issue is resolved though, it is a Brave browser issue.

**Reply by RadimKotajnyDev** [reply]
No, and I think it's resolved. My website is deployed on Vercel and I don't see that error anymore. Thanks a lot for help. 😄

**Reply by icyJoseph** [reply]
Nice. Pleas don't forget to mark as answered.
